### DISCUSSION TITLE
How to clean up Event logging because it takes lot of storage in the EC2 instant

### DISCUSSION THREAD

**Discussion by LeoDiep** [apache/superset#23110] [Q&A / Help]
Hi everyone, 
My record in event logs reach > 1 million after 1 year running and it takes 7GB of storage now. How can I clean these up?

I connected to superset.db by sqlite and run DELETE FROM logs where id <2000. It still delete recent records but checking the UI Action Log, recent log still there.

Please help me. Thanks

**Comment by LeoDiep** [comment]
is there any advices on this, or how to reduce the size of superset.db.

Thank you guys

**Comment by sfirke** [comment]
Maybe a silly question but is the db you deleted from your production metadata db?  Like, is all of your reports data etc. stored in that sqlite db and not a Postgres or MySQL db elsewhere?

**Reply by LeoDiep** [reply]
Hi,
My database for dashboards, charts is located on AWS S3 and I am using AWS Athena to query data from that to Superset.  Also the Superset is running on an AWS EC2.

I was mentioning about the superset.db on /home/ubuntu/.superset/ where it storage it own metadata and other data such as logs,...
As recommended in this link https://github.com/apache/superset/issues/9748 I connected superset.db by sqlite but still don't know how to clean it.

**Reply by sfirke** [reply]
In Superset 2.0.0+ my logs are stored in the same database as dashboards & charts data, in a table called `logs`.  If you delete the records from that table on your AWS S3 database does it help?  If you have an S3 database with your application data, I don't see how SQLite fits into the picture.

**Reply by LeoDiep** [reply]
Hi, This is 2 databases of mine
![image](https://user-images.githubusercontent.com/99783557/228726222-a9e2f154-977c-4a8f-aa72-7099662907c3.png)

This is the logs that I want to clean
![image](https://user-images.githubusercontent.com/99783557/228726618-59dbfdf8-4f0e-46a2-a8f0-470d2d96c814.png)

**Reply by LeoDiep** [reply]
This is the connection to the superset.db
![image](https://user-images.githubusercontent.com/99783557/228726873-73eba7eb-5114-442c-901e-cc00681daac9.png)

I think my superset.db doesn't have any production data, but the size of it keep increasing. So I guess the logs is the one causing the problem and I would like to clean it

**Comment by apole** [comment]
Facing same issue, any way to set verbosity?

**Comment by sfirke** [comment]
Over time, most of a Superset instance's database storage usage will be in the `logs` table. E.g., my instance had something like 1.9 million events in the logs table which took up about 1.3 GB.

You may want this table to analyze usage of charts and dashboards; if so, back it up.

If not, you can trim this data by connecting to your metadata DB (probably Postgres or MySQL) and running:
```
DELETE FROM logs WHERE dttm < '2025-05-10'; -- adjust date as needed
```
